core/services/workflows/engine.go

```diff
@@ -786,7 +786,6 @@ func (e *Engine) workerForStepRequest(ctx context.Context, msg stepRequest) {
 	curStep, verr := e.workflow.Vertex(msg.stepRef)
 	if verr != nil {
 		l.Errorf("failed to resolve step in workflow; error %v", err)
-		return
```

Comment:
@patrickhuie19 You need to put the metrics in an else or skip emitting the capability ID; as it stands I think this will panic because you're acccessing .ID on curStep which presumably would be nil